### PR TITLE
os: Chmod: don't test on wasi yet, wasi-libc does not yet support it

### PR DESCRIPTION
**Pull request by dkegel-fastly**
Lets "tinygo test -target wasi os" compile, if not pass.

For https://github.com/tinygo-org/tinygo/issues/2369

### CODE REVIEW

**Review by dgryski — APPROVED** [review]
LGTM

File: src/os/os_chmod_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// +build !baremetal,!js,!wasi
+
+// TODO: Move this into os_test.go (as upstream has it) when wasi supports chmod
```
Comment (niaow) [inline src/os/os_chmod_test.go]: Should this keep the license header from `os_test.go`?

Comment (dkegel-fastly) [reply src/os/os_chmod_test.go]: I keep forgetting that...

**Review by niaow — APPROVED** [review]
LGTM. Will merge once CI finishes.

**Pull request merged**